Src/ST-LIB_LOW/StateMachine/StateMachine.cpp

```diff
@@ -269,11 +280,20 @@ void StateMachine::force_change_state(uint8_t new_state) {
 
 	unregister_all_timed_actions(current_state);
 	exit_state(current_state);
-
+	#ifdef SIM_ON
+		old_state=current_state;
+	#endif
```

Comment:
no need to store the previos state to search then for it, look at the other comments